src/score_hv/harvesters/gsi_satellite_radiance.py

```diff
@@ -0,0 +1,415 @@
+        
+        
+        # get the datetime from the input file name
+        self.datetime = datetime.strptime(
+            self.config.harvest_filename.split('.')[-1].split('_')[0],
+            '%Y%m%d%H')
+        
+        with open(self.config.harvest_filename, encoding="utf-8") as f:
+            self.lines = list(f)
+            
+        self.parse_fit_file()           
+        
+        for obs_type, series_num_list in self.obs_type_series_numbers.items():
+            for var in self.config.vars_to_harvest:            
+                
+                values_by_channel = list()
+                for channel_index, series_number in enumerate(series_num_list):
+                    """return data usage info (by channel)
+                    """
+                    assert obs_type == self.channels[series_number][
+                                           'observation_type']
+                    assert self.obs_type_channels[obs_type][
+                               channel_index] == self.channels[series_number][
+                                                     'channel']
+                    values_by_channel.append(
+                        self.channels[series_number]['data_usage_dict'][var]
+                    )
+                    
+                self.satinfo_stats.append(SatinfoStat(
+                    self.datetime,
+                    None,
+                    obs_type,
+                    series_num_list,
+                    self.obs_type_channels[obs_type],
+                    var, # name of statistic
+                    values_by_channel,
+                    None
+                    )
+                )
+        
+        for obs_type, series_num_list in self.obs_type_series_numbers.items():
+            for gsi_stage, channel_stats_dict in self.channel_stats.items():
+                if len(series_num_list & channel_stats_dict.keys()) > 0:
+                    """this obs_type has active channels
+                    return stats by channel for this obs_type
+                    """
+                    for stat in self.config.stats_to_harvest:
+                        values_by_channel = list()
+                        longnames = list()
+                        for channel_index, series_number in enumerate(
+                                                               series_num_list):
+                            if series_number in channel_stats_dict.keys():
+                                """active channel!
+                                """
+                                channel_stat_dict = channel_stats_dict[
+                                                            series_number][stat]
+                                
+                                assert obs_type == channel_stat_dict[
+                                                             'observation_type']
+                                assert self.obs_type_channels[
```

Comment:
For these four assert statements, how likely are they to fail? 
If they fail, should they crash the code completely?
If they are likely to fail, it seems like custom exceptions would be more appropriate to provide the user with a clearer understanding of why the code crashed.

Reply:
all assert statements in this file have either been removed or replaced with if blocks -> raise ValueErrors